WIP: Refactor m.m.event to follow the core patterns

gkleiman opened the pull request.

aquamatthias commented on src/main/scala/mesosphere/marathon/api/MarathonRestModule.scala:
```diff
@@ -7,7 +7,7 @@ import com.google.inject.servlet.ServletModule
 import mesosphere.chaos.http.HttpConf
 import mesosphere.marathon.MarathonConf
-import mesosphere.marathon.event.http.HttpEventStreamServlet
+import mesosphere.marathon.core.event.impl.stream.HttpEventStreamServlet
```
`impl` should be a module private package. Importing it outside would violate this. Any ideas? (e.g. treat it as Servlet)

aquamatthias commented on src/main/scala/mesosphere/marathon/api/v2/EventSubscriptionsResource.scala:
```diff
@@ -23,18 +23,16 @@ import scala.concurrent.Future
-
-  @Inject(optional = true) val service: HttpCallbackSubscriptionService = null
+    val authorizer: Authorizer,
+    val maybeService: Option[HttpCallbackSubscriptionService]) extends AuthResource {
```
Instead of having an `Option[HttpCallbackSubscriptionService]` can we use `HttpCallbackSubscriptionService` with a different implementation (e.g. noop)? So client code does not have to deal with the existence of the service?

aquamatthias commented on src/test/scala/mesosphere/marathon/api/v2/EventSubscriptionsResourceTest.scala:
```diff
@@ -2,7 +2,7 @@ package mesosphere.marathon.api.v2
 import mesosphere.marathon._
 import mesosphere.marathon.api.TestAuthFixture
-import mesosphere.marathon.event.http.HttpCallbackSubscriptionService
+import mesosphere.marathon.core.event.impl.http.HttpCallbackSubscriptionService
```
Perhaps extract a trait for this class?

aquamatthias commented:
@gkleiman Thanks! Some violations, that we should fix. Otherwise LGTM.

gkleiman commented:
@aquamatthias thanks for the review!

I addressed the comments and moved some stuff around.

I think that maybe `mesosphere.marathon.core.events.history` should be an independent module (`mesosphere.marathon.core.history`).

aquamatthias commented on src/main/scala/mesosphere/marathon/core/event/http/impl/ActorHttpCallbackSubscriptionService.scala:
```diff
@@ -0,0 +1,30 @@
+
+import scala.concurrent.Future
+
+class ActorHttpCallbackSubscriptionService(
```
To be consistent, this class should be called: `HttpCallbackSubscriptionServiceDelegate`

aquamatthias replied:
The Actor should be called `HttpCallbackSubscriptionServiceActor`

aquamatthias commented:
@gkleiman 
1) I agree history has nothing to do with delivery of events and should be a separate component. Module is missing here.
2) Now you have a mixed setup: 1 conf and module but 2 different component packages (http, event). I would suggest: leaving this one component (1 conf/module) and having 2 packages under impl (e.g. callback, stream)

aquamatthias commented on src/main/scala/mesosphere/marathon/core/event/EventModule.scala:
```diff
@@ -0,0 +1,93 @@
+class EventModule(
+    eventBus: EventStream,
+    actorSystem: ActorSystem,
+    conf: MarathonConf,
```
Should be EventConf

gkleiman replied:
`HttpEventStreamServlet` needs stuff from `MarathonConf`.

aquamatthias replied:
It only reads conf.zkTimeoutDuration - it does not need the whole MarathonConf


> Am 18.07.2016 um 18:22 schrieb Gastón Kleiman <notifications@github.com>:
> 
> In src/main/scala/mesosphere/marathon/core/event/EventModule.scala <https://github.com/mesosphere/marathon/pull/4104#discussion_r71179493>:
> 
> > +import mesosphere.marathon.core.base.Clock
> > +import mesosphere.marathon.core.election.ElectionService
> > +import mesosphere.marathon.core.event.impl.callback._
> > +import mesosphere.marathon.core.event.impl.stream._
> > +import mesosphere.marathon.metrics.Metrics
> > +import mesosphere.marathon.plugin.auth.{ Authenticator, Authorizer }
> > +import mesosphere.marathon.state.EntityStore
> > +import org.eclipse.jetty.servlets.EventSourceServlet
> > +import org.slf4j.LoggerFactory
> > +
> > +import scala.concurrent.ExecutionContext
> > +
> > +class EventModule(
> > +    eventBus: EventStream,
> > +    actorSystem: ActorSystem,
> > +    conf: MarathonConf,
> HttpEventStreamServlet needs stuff from MarathonConf.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/mesosphere/marathon/pull/4104/files/2d3755c8a8c7e743ecdb026464f974daa07f70e2#r71179493>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAUK9eFcF86dRV_KH6CLio7Skn-EyxIFks5qW6hSgaJpZM4JNmzw>.
>

aquamatthias commented on src/main/scala/mesosphere/marathon/core/history/HistoryConf.scala:
```diff
@@ -0,0 +1,15 @@
+
+trait HistoryConf extends ScallopConf {
+  //scalastyle:off magic.number
+  lazy val eventStreamMaxOutstandingMessages = opt[Int](
```
That should be part of EventConf

aquamatthias commented on src/main/scala/mesosphere/marathon/core/event/EventModule.scala:
```diff
@@ -0,0 +1,93 @@
+
+import scala.concurrent.ExecutionContext
+
+class EventModule(
```
Would be nice to see a little bit of documentation:
- what is the responsibility of this component
- what does this component provide

aquamatthias commented on src/main/scala/mesosphere/marathon/core/event/EventModule.scala:
```diff
@@ -0,0 +1,93 @@
+    authorizer: Authorizer) {
+  val log = LoggerFactory.getLogger(getClass.getName)
+
+  private lazy val httpCallbacksEnabled: Boolean = {
```
Should be part of config

aquamatthias commented on src/main/scala/mesosphere/marathon/core/event/EventModule.scala:
```diff
@@ -0,0 +1,90 @@
+class EventModule(
+    eventBus: EventStream,
+    actorSystem: ActorSystem,
+    conf: MarathonConf,
```
Still think this should be EventConf

aquamatthias commented:
@gkleiman Thanks!
